internal/vds/axis.h

```diff
@@ -0,0 +1,39 @@
+#ifndef AXIS_H
+#define AXIS_H
+
+#include <memory>
+#include <string>
+
+#include <OpenVDS/OpenVDS.h>
+
+#include "vds.h"
+
+class Axis {
+public:
+    Axis(
+        enum api_axis_name const                api_name,
+        OpenVDS::VolumeDataLayout const * const layout
+    );
+
+    int get_number_of_points() const noexcept(true);
```

Comment:
* `point` -> `sample` 👍 
* I am happy to change it to the shorter naming convention if that is what we want. :)
* I ended up up prefixing getter with `get_` here because I get a name collision between functions like `int index()` and the member `this->index`. I decided to go with the `get_` prefix now, but I am also happy to change this. Then there is the question if we find a good way to have different names for members and getters or if we want to prefix/postfix the member variable.

Reply:
I would be fine with post-fixing member variables with an underscore. After all it's what's on the outside that matters, not the inside. Keep the outside pretty! ... I learned that on keeping up with the Kardashians 😆

Reply:
We can also go for something like `m_index` if we consider the better in terms of readability.

Reply:
Since I'm enforcing the use of `this->` anyway I would opt for a trailing underscore. But I'm fine with both